spim_core/spim_base.py

```diff
@@ -112,6 +114,95 @@ def log_to_file(self, filepath: Path, logger: Logger = None,
+        """Checks ext disk space before scan to see if disk has enough space scan
+
+        :param xtiles: number of x tiles
+        :param ytiles: number of y tiles
+        :param ztiles: number of z tiles
+        """
+        # One tile (tiff) is ~10368 kb
+        est_stack_filesize = self.cfg.bytes_per_image * ztiles
+        est_scan_filesize = est_stack_filesize*xtiles*ytiles
+        if est_scan_filesize >= shutil.disk_usage(self.cfg.ext_storage_dir).free:
+            self.log.error("Not enough space in external directory")
+            raise
+
+    def check_local_disk_space(self, z_tiles):
+        """Checks local disk space before scan to see if disk has enough space for two stacks
+
+           :param z_tiles: number of z tiles
+        """
+
+        est_filesize = self.cfg.bytes_per_image * z_tiles
+        if est_filesize * 2 >= shutil.disk_usage(self.cfg.local_storage_dir).free:
+            self.log.error("Not enough space on disk. Is the recycle bin empty?")
+            raise
+
+    def check_read_write_speeds(self, drive: Path):
+        """Check local read/write speeds to make sure it can keep up with acquisition
+
+        :param drive: drive you are testing read/write speeds. Usually the local or external storage of instrument
+        """
+
+        test_filename = fr"{drive}\test.txt"
+        f = open(test_filename, 'a')  # Create empty file to check reading/writing speed
+        f.close()
+        try:
+            speed_MB_s = {}
+            for check in ['read', 'write']:
+                output = subprocess.check_output(
+                    fr'fio --name=test --filename={test_filename} --size=16Gb --rw={check} --bs=1M '
+                    r'--direct=1 --numjobs=1 --ioengine=windowsaio --iodepth=1 --runtime=0 --startdelay=0 '
+                    r'--thread --group_reporting', shell=True)
+                out = str(output)
+                speed_MB_s[check] = round(float(out[out.find('BW=') + len('BW='):out.find('MiB/s')])/(10**6/2**20))
+
+
+            # converting B/s to MB/s
+            acq_speed_MB_s = (self.cfg.bytes_per_image*(1/1000000)) * (1/self.cfg.get_period_time())
+            # Delete test file
+            os.remove(test_filename)
+
+            # Go through both speeds and specify if one or both are the problem
+            read_too_slow = False
+            write_too_slow = False
+
+            if speed_MB_s['read'] <= acq_speed_MB_s:
+                read_too_slow = True
+                self.log.warning(f'{drive} read speed too slow')
+
+            if speed_MB_s['write'] <= acq_speed_MB_s:
+                write_too_slow = True
+                self.log.warning(f'{drive} write speed too slow')
+
+            if read_too_slow or write_too_slow:
+                raise
+        except subprocess.CalledProcessError:
+            self.log.warning('fios not installed on computer. Cannot verify read/write speed')
+
+    def _check_system_memory_resources(self, channel_count: int,
```

Comment:
Is there a reason this function and not the others are _function()?

Reply:
Sonya wrote this one so I'm not entirely sure her reasoning on making this function private